### ISSUE TITLE
Support fractional values for the pixel scale multiplier

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#8608]
With 1.18, Wesnoth is able to do integer scaling (1x, 2x, 3x, 4x). 1x is generally for 1080p and smaller while 2x is for 4K monitors (and 3x and 4x essentially don't matter), which leaves a big gap between 1080p and 4K that is still too small when at 1x scaling but too large at 2x scaling. So being able to support at least 1.25x and 1.5x scaling multipliers would be beneficial.

**Comment by Wedge009** [comment]
Agreed, even having recently 'graduated' from QHD (1440p) to UHD (2160p), I find 2x too big. I use 1.5x in the desktop environment, but Wesnoth is independent of that.

Completely understand this is a tricky enhancement, though.

**Comment by doofus-01** [comment]
As discussed [here](https://github.com/wesnoth/wesnoth/commit/5c5167cba0f65d173b812250f3a1e737c76e6c09), simply changing the threshold for automatic pixel scale multiplier seems to help avoid stumbling into 2X with a window that is something like 1922x1081.  The help portraits are one issue, but the dialogs are also full of scrollbars.  The map shouldn't be negatively impacted because you can still zoom in/out to something almost unusable either way.

To test, try changing ` wesnoth/src/preferences/general.cpp#L72-L73` to 2560x1440.
(It feels silly to open a PR for that as long as there is any chance of discussion)

It's not a real fix, but I think mitigates the worst part of the issue.

**Comment by Pentarctagon** [comment]
I think these are separate issues - it'd be a good change to prevent any resolution over 1080p from switching to 2x pixel scaling (and a PR adjusting those values would be fine), however this is about the issue of wanting intermediate scaling options between 1x and 2x.
